### PR TITLE
fix(checker): allow whitespace before inner functions and class

### PR DESCRIPTION
**Pull request by johanfleury**
Changes D202: ("No blank lines allowed after function docstring") to
allow space below function docstrings with inner functions.

i.e. allows:

```python
def outer():
    """Valid docstring."""

    def inner():
        pass

    return pass
```

See comment from @cdeil in #361.

---

This is a follow-up of PR #395.

---

- [x] Add unit tests and integration tests where applicable.  
      If you've added an error code or changed an error code behavior,
      you should probably add or change a test case file under `tests/test_cases/` and add 
      it to the list under `tests/test_definitions.py`.  
      If you've added or changed a command line option,
      you should probably add or change a test in `tests/test_integration.py`.
- [x] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".  
      Make sure to include the PR number after you open and get one.

### CODE REVIEW

**Comment by Code0x58** [comment]
I just thought, there is the case of decorators around classes/functions, so that regex could use and test file could use a `@` case

**Comment by johanfleury** [comment]
I updated the PR and also added a commit to fix some `DeprecationWarning`.

**Comment by ColinKennedy** [comment]
I just stumbled upon this PR. Is there anything that I can do to help it along or is it ready to merge?

**Comment by johanfleury** [comment]
@ColinKennedy we're just waiting for a review by a project member.

**Comment by Nurdok** [comment]
Sorry for the long delays in response - I've been on back-to-back family vacation and business trips. I'll to catch up as soon as I can.

File: src/pydocstyle/checker.py [hunk 1]
```diff
@@ -198,7 +198,11 @@ def check_no_blank_before(self, function, docstring):  # def
             if blanks_before_count != 0:
                 yield violations.D201(blanks_before_count)
             if not all(blanks_after) and blanks_after_count != 0:
-                yield violations.D202(blanks_after_count)
+                if not (
```
Comment (Nurdok) [inline src/pydocstyle/checker.py]: Add a comment explaining this, possibly defining the regex in a constant or local variable with a descriptive name.

Comment (johanfleury) [reply src/pydocstyle/checker.py]: I completed function's docstring and added a comment on the if statement as well. I didn't find a good/useful way to name the regex so I kept it as it was.

**Pull request merged**